dangerfile.js

```diff
@@ -0,0 +1,181 @@
+  const tableHeaders = [headers.join(' | '), headers.map(() => ' --- ').join(' | ')];
+
+  const tablebody = body.map(r => r.join(' | '));
+  return `${tableHeaders.join('\n')}\n${tablebody.join('\n')}`;
+}
+
+function generateEmphasizedChange([bundle, { parsed, gzip }]) {
+  // increase might be a bug fix which is a nice thing. reductions are always nice
+  const changeParsed = addPercent(parsed.relativeDiff, ':heart_eyes:', '');
+  const changeGzip = addPercent(gzip.relativeDiff, ':heart_eyes:', '');
+
+  return `**${bundle}**: parsed: ${changeParsed}, gzip: ${changeGzip}`;
+}
+
+async function run() {
+  // Use git locally to grab the commit which represents the place
+  // where the branches differ
+  const upstreamRepo = danger.github.pr.base.repo.full_name;
+  const upstreamRef = danger.github.pr.base.ref;
+  try {
+    await git(`remote add ${upstreamRemote} https://github.com/${upstreamRepo}.git`);
+  } catch (err) {
+    // ignore if it already exist for local testing
+  }
+  await git(`fetch ${upstreamRemote}`);
+  const mergeBaseCommit = await git(`merge-base HEAD ${upstreamRemote}/${upstreamRef}`);
+
+  const commitRange = `${mergeBaseCommit}...${danger.github.pr.head.sha}`;
+
+  const comparison = await loadComparison(mergeBaseCommit, upstreamRef);
+  const results = Object.entries(comparison.bundles);
+  const anyResultsChanges = results.filter(createComparisonFilter(1, 1));
+
+  if (anyResultsChanges.length > 0) {
+    markdown('This PR introduced some changes to the bundle size.');
+
+    const importantChanges = results
+      .filter(createComparisonFilter(parsedSizeChangeThreshold, gzipSizeChangeThreshold))
+      .filter(isPackageComparison)
+      .map(generateEmphasizedChange);
+
+    // have to guard against empty strings
+    if (importantChanges.length > 0) {
+      markdown(importantChanges.join('\n'));
+    }
+
+    const detailsTable = generateMDTable(
+      [
+        'bundle',
+        'parsed diff',
+        'gzip diff',
+        'prev parsed',
+        'current parsed',
+        'prev gzip',
+        'current gzip',
+      ],
+      results.map(([bundle, { parsed, gzip }]) => {
+        return [
+          bundle,
+          addPercent(parsed.relativeDiff),
+          addPercent(gzip.relativeDiff),
+          parsed.previous,
+          parsed.current,
+          gzip.previous,
+          gzip.current,
+        ];
+      }),
+    );
+
+    const details = `
+  <details>
+  <summary>Details of bundle changes.</summary>
+
+  <p>Comparing: ${commitRange}</p>
+
+  ${detailsTable}
+
+  </details>`;
+
+    markdown(details);
+  } else {
+    // this can later be removed to reduce PR noise. It is kept for now for debug
+    // purposes only. DangerJS will swallow console.logs if completes successfully
+    markdown(`No bundle size changes comparing ${commitRange}`);
+  }
+}
+
+(async () => {
+  try {
+    await run();
+  } catch (err) {
+    console.error(err);
+    // need to cleanup first so no early exit
```

Comment:
So we lose the exit code 1?

Reply:
I need to check this again. The bot will leave a comment with the fail which might be confusing (following react rationale). But then the CI task will pass and therefore the build will be green. Maybe I can get the job to turn red while the bot stays silent.